### PR TITLE
Change exception throwing in Path.GetRelativePath

### PR DESCRIPTION
**Pull request by Marusyk**
Related to #38197

Please review.
Thank you in advance

### CODE REVIEW

**Comment by Wraith2** [comment]
The CoreCLR PR is https://github.com/dotnet/coreclr/pull/25047
I think the change to the tests belongs here but the GetRelativePath change should just be in CoreCLR. Once merged in CoreCLR there will be an update at some point to pull in shared code from CoreCLR into CoreFX and after that this PR will be mergable.

This will also require a change to the documentation, https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getrelativepath?view=netcore-3.0 to show that it can now throw the new exception type. I don't know how you make sure this happens.

File: src/Common/src/CoreLib/System/IO/Path.cs [hunk 1]
```diff
@@ -812,8 +812,12 @@ public static string GetRelativePath(string relativeTo, string path)
...
         private static string GetRelativePath(string relativeTo, string path, StringComparison comparisonType)
         {
-            if (string.IsNullOrEmpty(relativeTo)) throw new ArgumentNullException(nameof(relativeTo));
-            if (PathInternal.IsEffectivelyEmpty(path.AsSpan())) throw new ArgumentNullException(nameof(path));
+            if (relativeTo == null)
+                throw new ArgumentNullException(nameof(relativeTo));
+            if (relativeTo.Length == 0)
+                throw new ArgumentException(nameof(relativeTo));
+            if (PathInternal.IsEffectivelyEmpty(path.AsSpan()))
+                throw new ArgumentNullException(nameof(path));
```
Comment (stephentoub) [inline src/Common/src/CoreLib/System/IO/Path.cs]: Please remove this.  This change needs to be made in the dotnet/coreclr repo.  This code is mirrored but isn't actually compiled in corefx.

File: src/System.Runtime.Extensions/tests/System/IO/Path.GetRelativePath.cs [hunk 2]
```diff
@@ -148,5 +148,12 @@ public static void GetRelativePath_Mac(string relativeTo, string path, string ex
                 ignoreLineEndingDifferences: false,
                 ignoreWhiteSpaceDifferences: false);
         }
+
+        [Fact]
+        public static void GetRelativePath_InvalidArgs()
+        {
+            Assert.Throws<ArgumentNullException>(() => Path.GetRelativePath(null, "."));
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/IO/Path.GetRelativePath.cs]: Can this validate the parameter name as well?

**Comment by danmosemsft** [comment]
@stephentoub looks like your feedback was addressed.

**Comment by stephentoub** [comment]
> looks like your feedback was addressed.

@danmosemsft, yes, but the coreclr PR it depends on hasn't been merged.

**Comment by danmosemsft** [comment]
@stephentoub i missed that 🙂

**Comment by Marusyk** [comment]
Hello, the PR in coreclr has been merged

**Comment by danmosemsft** [comment]
CI not working?

**Comment by danmosemsft** [comment]
@dotnet/dnceng can you help get CI to work?

**Comment by Chrisboh** [comment]
@danmosemsft it looks like CI ran here and everything finished in an hour.  What exactly was the issue you were seeing?

**Comment by danmosemsft** [comment]
@chrisboh if I click on one of the failed legs, and go to azure pipelines, it says "Build not found". But if I try to restart, it doesn't help.

**Comment by alexperovich** [comment]
The issue here is that AzDev has a bug that is causing old, deleted legs to show up as checks. If you click the "corefx-ci" check for the entire build it shows a build with one failed test work item.

**Comment by alexperovich** [comment]
Actually, I see that this commit https://github.com/dotnet/corefx/commit/519eccdf8e9c8cb0390f91da8561fcf98a19b838 changed the names of some of the legs. The first run of this PR ran without that change so it has old checks.

**Comment by danmosemsft** [comment]
Yeah, @chrisboh just explained. There was one real test failure, but it's unrelated. thx!

**Comment by danmosemsft** [comment]
Thanks @Marusyk apologies for the delay. We will try to move more quickly with any future PR's

**Pull request merged**